modApiServer/src/org/aion/api/server/rpc3/PersonalRPC.java

```diff
@@ -18,29 +21,27 @@
     public String execute(String requestString){
         try{
 
-        Request request = RequestConverter.decode(requestString);
-        //check that the request can be fulfilled by this class
-        String interfaceName = request.method.split("_")[0];
-        if(interfaceName.equals("personal")){
+            Request request = RequestConverter.decode(requestString);
+            //check that the request can be fulfilled by this class
+            String interfaceName = request.method.split("_")[0];
+            if(interfaceName.equals("personal")){
                 if(request.method.equals("personal_ecRecover")){
                     EcRecoverParams params=EcRecoverParamsConverter.decode(request.params);
 
-                    String result = personal_ecRecover(params.dataThatWasSigned,params.signature);
-                    return DataHexStringConverter
-                            .encode(result);
+                    ByteArrayWrapper result = personal_ecRecover(params.dataThatWasSigned,params.signature);
+                    return DataHexStringConverter.encode(result);
                 }
                 else
                     throw new MethodNotFoundRPCException();
-        }
-        else{
-            throw new InternalErrorRPCException();
-        }
+            }
+            else{
+                throw new InternalErrorRPCException();
+            }
         }
         catch(InvalidRequestRPCException |ParseErrorRPCException |MethodNotFoundRPCException |InvalidParamsRPCException |InternalErrorRPCException e){
             return e.getMessage();
         }
-
     }
 
-    protected abstract String personal_ecRecover(String dataThatWasSigned,String signature);
-}
\ No newline at end of file
+    protected abstract ByteArrayWrapper personal_ecRecover(String dataThatWasSigned,ByteArrayWrapper signature);
```

Comment:
Not sure why to use ByteArrayWrapper? Is any reason we can not use byte[]?
The ByteArrayWrapper is mainly for map pair comparing due to the map can not compare the byte[] properly.